### PR TITLE
5.x Add a local development environment for running tests

### PR DESCRIPTION
**Pull request by markstory**
I've used this docker-compose environment to run tests locally and they all passed with 4.x. I'll use this to get the tests passing against cake5.

### CODE REVIEW

**Comment by othercorey** [comment]
Looks ok, but some other cleanup is needed.

File: Dockerfile [hunk 1]
```diff
@@ -15,10 +21,16 @@ RUN apt-get update && \
 
 RUN LC_ALL=C.UTF-8 add-apt-repository ppa:ondrej/php && \
   apt-get update && \
-  apt-get install -y php7.2-cli php7.2-mbstring php7.2-xml php7.2-zip php7.2-intl php7.2-opcache php7.2-sqlite
+  apt-get install -y \
+    php8.1-cli php8.1-mbstring php8.1-xml php8.1-zip php8.1-intl php8.1-opcache php8.1-sqlite php8.1-curl \
```
Comment (othercorey) [inline Dockerfile]: would you mind splitting this into 1 package per line?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -68,34 +68,34 @@ jobs:
...
 
     - name: Submit code coverage
-      if: matrix.php-version == '7.4'
+      if: matrix.php-version == '8.1'
       uses: codecov/codecov-action@v1
 
   cs-stan:
     name: Coding Standard & Static Analysis
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-21.10
```
Comment (othercorey) [inline .github/workflows/ci.yml]:
```suggestion
    runs-on: ubuntu-20.04
```

**Comment by markstory** [comment]
Merging this. I'll have more pull requests coming in the future for typehints and I'm planning on removing the usage of QueryTrait so that we can also remove it from cake core.

**Pull request merged**